include/polo/utility/custom_termination.hpp

```diff
@@ -0,0 +1,30 @@
+#ifndef POLO_UTILITY_CUSTOMTERM_HPP_
+#define POLO_UTILITY_CUSTOMTERM_HPP_
+
+namespace polo {
+namespace utility {
+namespace terminator {
+template <class value_t, class index_t> struct custom {
+  using func_t = bool *(*)(const index_t, const value_t, const value_t *,
+                           const value_t *, const value_t *, void *);
+
+  custom(func_t terminate, void *data) {
+    terminate_ = terminate;
+    data_ = data;
+  }
+
+  template <class InputIt1, class InputIt2>
```

Comment:
template parameters should not be there --- anyways `terminate_` works on pointers with the assumption of having contiguous memory locations.